src/util/as_const.h

```diff
@@ -16,4 +16,8 @@ const T &as_const(T &value)
   return static_cast<const T &>(value);
 }
 
+/// Deleted to avoid calling as_const on a rvalue
```

Comment:
Nitpick: this is an `xvalue` not an `rvalue` -- see https://en.cppreference.com/w/cpp/language/value_category